pkg/update/rolling_restart.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020 The Cockroach Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package update
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/errors"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/apps/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO this is not working.
+// We need to add an annotation the statefulset with a value that will
+// cause the pods to roll.
+// See https://github.com/kubernetes/kubernetes/blob/bd4d197b5267ee198c1b0a070d7398f10df68c52/staging/src/k8s.io/kubectl/pkg/polymorphichelpers/objectrestarter.go#L95
+
+// RollingRestart performs a rolling restart on a sts.
+func RollingRestart(
```

Comment:
Should this be commented out or removed until it's working?

Reply:
Commented, but the code works, but it does not roll the cluster because we are not patching the STS.  We patch the sts and this code will work.